docs/components/table-view.md

```diff
@@ -347,6 +352,74 @@ const getRowLink = (row: Record<string, any>): RowLink => ({
 </script>
 ```
 
+### rowBulkAction
+
+Function for making selected rows unavailable for bulk actions selection conditionally. The function receives row value object as an argument and should return a boolean or an object with two parameters:
+
+```ts
+type RowBulkAction = boolean | { disabled: boolean, disabledTooltip?: string }
+```
+
+When function returns a boolean `true`, a row will be enabled. Otherwise, when `false` is returned, it will be disabled. Alternatively, it can return an object (helpful when you need to display a tooltip explaining why the row is unavailable for bulk actions selection).
+
+<KTableView
+  :row-bulk-action="getRowBulkAction"
+  :data="paginatedData"
+  :headers="basicHeaders(false, null, null, true)"
+  :pagination-attributes="{ totalCount: basicPaginatedData.length, pageSizes: [5, 10] }"
+  @page-change="onPageChange"
+  @page-size-change="onPageSizeChange"
+  @bulk-actions-select="onBulkActionsSelect"
+>
+  <template #bulk-action-items>
+    <KDropdownItem danger>
+      Delete ({{ selectedData.length }} items)
+    </KDropdownItem>
+  </template>
+</KTableView>
+
+```vue
+<template>
+  <KTableView
+    :row-bulk-action="getRowBulkAction"
+    :data="paginatedData"
+    :headers="headers"
+    :pagination-attributes="{ totalCount: tableData.length, pageSizes: [5, 10] }"
+  >
+    <template #bulk-action-items>
+      <KDropdownItem danger>
+        Delete ({{ selectedData.length }} items)
+      </KDropdownItem>
+    </template>
+  </KTableView>
+</template>
+
+<script setup lang="ts">
+import type { RowBulkAction } from '@kong/kongponents'
+
+...
+
+const getRowBulkAction = (data: Record<string, any>): RowBulkAction => {
+  if (data.id === 2) {
+    return false
+  }
+
+  if (data.id === 3) {
+    return { disabled: true }
+  }
+
+  if (data.id === 4) {
+    return {
+      disabled: true,
+      disabledTooltip: 'This row is disabled.',
+    }
+  }
+
+  return true
+}
+</script>
```

Comment:
Since we reference `row` as the parameter, let's use that parameter here in the example
```suggestion
const getRowBulkAction = (row: Record<string, any>): RowBulkAction => {
  if (row.id === 2) {
    return false
  }

  if (row.id === 3) {
    return { disabled: true }
  }

  if (row.id === 4) {
    return {
      disabled: true,
      disabledTooltip: 'This row cannot be selected.',
    }
  }

  return true
}
</script>
```